api/src/controllers/events.ts

```diff
@@ -1 +1,52 @@
-// events.ts
+import { errors as elasticErrors } from "@elastic/elasticsearch";
+import { RequestHandler } from "express";
+import asyncHandler from "express-async-handler";
+import { Clients, Displayable } from "../types";
+import { PaginationQueryParameters } from "./pagination";
+import { Config } from "../../config";
+import { HttpError } from "./error";
+import { ResultList } from "../types/responses";
+import { resultListResponse } from "../helpers/responses";
+
+type QueryParams = {
+  query?: string;
+  sort?: string;
+  sortOrder?: string;
+  aggregations?: string;
+  "contributors.contributor"?: string;
+  "publicationDate.from"?: string;
+  "publicationDate.to"?: string;
+  format?: string;
+} & PaginationQueryParameters;
+
+type EventsHandler = RequestHandler<never, ResultList, never, QueryParams>;
+
+const eventsController = (clients: Clients, config: Config): EventsHandler => {
+  const index = config.eventsIndex;
+  const resultList = resultListResponse(config);
+
+  return asyncHandler(async (req, res) => {
+    const { query: queryString, ...params } = req.query;
+    try {
+      const searchResponse = await clients.elastic.search<Displayable>({
+        index,
+        _source: ["display"],
+      });
+
+      res.status(200).json(resultList(req, searchResponse));
+    } catch (error) {
```

Comment:
This is all very far in my memory, but looking at the `articles` controller, there is [a little bit more done in the error handling](https://github.com/wellcomecollection/content-api/blob/main/api/src/controllers/articles.ts#L136) which might be useful to this endpoint as well. 

And remind me, I think it was said in a meeting, are we skipping pagination for now?

Reply:
Yes! Will get this a bit more fleshed out now to match `articles` better. 

I thought we were doing pagination - I've just realised this isn't working.